shared/src/main/scala/tastyquery/ast/Types.scala

```diff
@@ -66,24 +68,91 @@ object Types {
 
   abstract class Type {
 
-    final def paramInfos: List[Type] = this match
+    /*final def paramInfos: List[Type] = this match
       case tpe: MethodType => tpe.paramTypes
       case tpe             => Nil
 
     final def tparamBounds: List[TypeBounds] = this match
       case tpe: PolyType => tpe.paramTypeBounds
-      case tpe           => Nil
+      case tpe           => Nil*/
 
     final def tparamRefs: List[TypeParamRef] = this match
-      case tpe: PolyType => tpe.refs
-      case tpe           => Nil
+      case tpe: TypeLambdaType => tpe.paramRefs
+      case tpe                 => Nil
+
+    /** The type parameters of this type are:
+      * For a ClassInfo type, the type parameters of its class.
+      * For a typeref referring to a class, the type parameters of the class.
+      * For a refinement type, the type parameters of its parent, dropping
+      * any type parameter that is-rebound by the refinement.
+      */
+    final def typeParams(using Context): List[ParamInfo.TypeParamInfo] =
```

Comment:
this should probably be internal maybe

Reply:
In dotty this looks like a utility for just computing type applications so I guess it should not be so visible